code/game/objects/items/food/pizza.dm

```diff
@@ -1,187 +1,196 @@
 
 /////////////////////////////////////////////////PIZZA////////////////////////////////////////
 
-/obj/item/reagent_containers/food/snacks/pizza
+/obj/item/food/pizza
 	icon = 'icons/obj/food/pizzaspaghetti.dmi'
 	w_class = WEIGHT_CLASS_NORMAL
-	slices_num = 6
-	volume = 80
-	list_reagents = list(/datum/reagent/consumable/nutriment = 28, /datum/reagent/consumable/nutriment/protein = 3, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
+	max_volume = 80
+	food_reagents = list(/datum/reagent/consumable/nutriment = 28, /datum/reagent/consumable/nutriment/protein = 3, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
 	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
-	foodtype = GRAIN | DAIRY | VEGETABLES
+	foodtypes = GRAIN | DAIRY | VEGETABLES
 
-/obj/item/reagent_containers/food/snacks/pizzaslice
+/obj/item/food/pizza/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice, 6, 30)
+
+/obj/item/food/pizzaslice
 	icon = 'icons/obj/food/pizzaspaghetti.dmi'
-	list_reagents = list(/datum/reagent/consumable/nutriment = 5)
-	foodtype = GRAIN | DAIRY | VEGETABLES
-	value = FOOD_JUNK
+	food_reagents = list(/datum/reagent/consumable/nutriment = 5)
+	foodtypes = GRAIN | DAIRY | VEGETABLES
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/food/pizzaslice/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_ROLLINGPIN, /obj/item/stack/sheet/pizza, 1, 10)
```

Comment:
Since that's all subtypes defining a slice type var would be better here instead of repeating this.

Reply:
if he has to do that he might as well do it on /food level

Reply:
just looked at the code again and I realized: no this shouldn't be a slice var because you might have multiple processing recipes and this keeps it easier to manage than a slice var.